app/src/main/java/com/easyrouteapp/fragment/RoutesFragment.java

```diff
@@ -0,0 +1,95 @@
+package com.easyrouteapp.fragment;
+
+
+import android.os.Bundle;
+import android.support.v4.widget.SwipeRefreshLayout;
+import android.support.v7.widget.LinearLayoutManager;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import com.easyrouteapp.R;
+import com.easyrouteapp.adapter.RouteAdapter;
+import com.easyrouteapp.component.CustomRecycleView;
+import com.easyrouteapp.domain.Route;
+import com.easyrouteapp.event.LoadDataServiceErrorEvent;
+import com.easyrouteapp.event.ReturnLoadDataEvent;
+
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RoutesFragment extends RefreshableFragment {
+    private static final String TAG_LOG = "[RoutesFragment]";
+
+    private CustomRecycleView routesRV;
+    protected SwipeRefreshLayout mSwipeRefreshLayout;
+    private View fragmentView;
+
+    public RoutesFragment() {
+        // Required empty public constructor
+    }
```

Comment:
In this case we need a public no-argument constructor because android fragment manager use this to re-instantiate the class when needed. If it isn't presente a run time exceltion will happen.